test/unit/hpa_background_thread.c

```diff
@@ -65,6 +65,23 @@ set_background_thread_enabled(bool enabled) {
 	expect_d_eq(0, err, "Unexpected mallctl failure");
 }
 
+static void
+wait_until_thread_is_enabled(unsigned arena_id) {
+	tsd_t* tsd = tsd_fetch();
+
+	bool sleeping = false;
+	int iterations = 0;
+	do {
+		background_thread_info_t *info =
+		    background_thread_info_get(arena_id);
+		malloc_mutex_lock(tsd_tsdn(tsd), &info->mtx);
+		malloc_mutex_unlock(tsd_tsdn(tsd), &info->mtx);
+		sleeping = background_thread_indefinite_sleep(info);
+		assert_d_lt(iterations, (int)1e6,
```

Comment:
It looks the `1e` syntax is part of the floating point literals? To be safe I'd say stick with `UINT64_C(1000000)` or so.